apps/linear_algebra/src/halide_blas.h

```diff
@@ -0,0 +1,225 @@
+#ifndef HALIDE_BLAS_H
+#define HALIDE_BLAS_H
+
+#include <cmath>
+
+#include "HalideRuntime.h"
+#include "halide_scopy_impl.h"
+#include "halide_dcopy_impl.h"
+#include "halide_sscal_impl.h"
+#include "halide_dscal_impl.h"
+#include "halide_saxpy_impl.h"
+#include "halide_daxpy_impl.h"
+#include "halide_sdot.h"
+#include "halide_ddot.h"
+#include "halide_sasum.h"
+#include "halide_dasum.h"
+#include "halide_sgemv_notrans.h"
+#include "halide_dgemv_notrans.h"
+#include "halide_sgemv_trans.h"
+#include "halide_dgemv_trans.h"
+#include "halide_sgemm_notrans.h"
+#include "halide_dgemm_notrans.h"
+
+inline int halide_scopy(buffer_t *x, buffer_t *y) {
+  return halide_scopy_impl(0, x, nullptr, y);
+}
+
+inline int halide_dcopy(buffer_t *x, buffer_t *y) {
+  return halide_dcopy_impl(0, x, nullptr, y);
+}
+
+inline int halide_sscal(float a, buffer_t *x) {
+  return halide_sscal_impl(a, x, nullptr, x);
+}
+
+inline int halide_dscal(double a, buffer_t *x) {
+  return halide_dscal_impl(a, x, nullptr, x);
+}
+
+inline int halide_saxpy(float a, buffer_t *x, buffer_t *y) {
+  return halide_saxpy_impl(a, x, y, y);
+}
+
+inline int halide_daxpy(double a, buffer_t *x, buffer_t *y) {
+  return halide_daxpy_impl(a, x, y, y);
+}
+
+inline int halide_sgemv(bool trans, float a, buffer_t *A, buffer_t *x, float b, buffer_t *y) {
+  if (trans) {
+    return halide_sgemv_trans(a, A, x, b, y, y);
+  } else {
+    return halide_sgemv_notrans(a, A, x, b, y, y);
+  }
+}
+
+inline int halide_dgemv(bool trans, double a, buffer_t *A, buffer_t *x, double b, buffer_t *y) {
+  if (trans) {
+    return halide_dgemv_trans(a, A, x, b, y, y);
+  } else {
+    return halide_dgemv_notrans(a, A, x, b, y, y);
+  }
+}
+
+inline int halide_sgemm(bool transA, bool transB, float a, buffer_t *A, buffer_t *B, float b, buffer_t *C) {
+  if (!transA && !transB) {
+    return halide_sgemm_notrans(a, A, B, b, C, C);
+  } else if (transA) {
+    // return halide_sgemv_trans_A(a, A, B, b, C, C);
```

Comment:
Do you mean to have these commented out? If so, additional comments as to why they are commented out are necessary.